pkg/backends/calico/client.go

```diff
@@ -107,7 +107,6 @@ func NewCalicoClient(confdConfig *config.Config) (*client, error) {
 		client:            bc,
 		cache:             make(map[string]string),
 		peeringCache:      make(map[string]string),
-		routeCache:        make(map[string]string),
```

Comment:
Perhaps controversial, but I removed the need for the `routeCache` here and instead have the RouteGenerator manage routes directly via new `Add` and `Delete` functions.

Since we're not blatting over the contents every time it changes, we don't need a separate cache any more.